slsim/Halos/halos.py

```diff
@@ -0,0 +1,758 @@
+        z = [z]
+
+    mass = []
+    if np.all(np.isnan(z)):
+        return [0] * len(z)
+    for z_val in z:
+        mass.append(
+            colossus_halo_mass_sampler(
+                m_min=m_min,
+                m_max=m_max,
+                resolution=resolution,
+                z=z_val,
+                cosmology=cosmology,
+                size=1,
+                sigma8=sigma8,
+                ns=ns,
+                omega_m=omega_m,
+            )
+        )
+    return mass
+
+
+def redshift_mass_sheet_correction_array_from_comoving_density(redshift_list):
+    """Create an array of redshifts for mass sheet correction calculations.
+
+    This utility function generates a sequence of redshift values beginning at 0.025 and
+    incrementing by 0.05, extending up to the maximum redshift found in the input
+    redshift_list. The generated redshift array is particularly useful for computations
+    related to mass sheet corrections in lensing studies, where discrete redshift values
+    are needed across a specified range.
+
+    :param redshift_list: Array of observed redshifts, the maximum of which determines
+        the upper limit for the generated redshift sequence. This list provides the
+        context or observational range within which the mass sheet correction
+        calculations are relevant.
+    :type redshift_list: ndarray
+    :return: An array of redshift values starting from 0.025 and increasing in steps of
+        0.05, terminating at or just below the maximum value found in redshift_list.
+        This structured approach ensures a consistent set of redshifts for subsequent
+        analytical steps.
+    :rtype: ndarray
+    """
+
+    z_max = redshift_list[-1]
+    linspace_values = np.arange(0.025, z_max, 0.05)
+    return linspace_values
+
+
+def kappa_ext_for_each_sheet(redshift_list, first_moment, sky_area, cosmology):
+    """Calculate the external convergence (kappa_ext) for lensing sheets at given
+    redshifts.
+
+    The function computes kappa_ext using the first moment of mass, the area under consideration,
+    and the critical surface mass density.
+
+    :param redshift_list: Redshifts of the lens planes for which kappa_ext is calculated.
+    :type redshift_list: list or ndarray
+    :param first_moment: First moment of mass for each redshift in the redshift_list.
+    :type first_moment: list or ndarray
+    :param sky_area: Area of the sky in square degrees under consideration for the lensing calculation.
+    :type sky_area: `~astropy.units.Quantity`
+    :param cosmology: Cosmology instance used for the calculation.
+    :type cosmology: astropy.cosmology instance
+    :return: Array of kappa_ext values for each redshift in the redshift_list.
+    :rtype: ndarray
+    """
+    cone_opening_angle = deg2_to_cone_angle(sky_area.value)
+    # TODO: make it possible for other geometry model
+
+    lens_cosmo = LensCosmo(z_lens=redshift_list, z_source=10, cosmo=cosmology)
```

Comment:
does this have implications that you set the source redshift =10? This might also be dangerous as you might use different conventions somewhere else (or there is at least the chance for that).